04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,22 @@
-#write your code here
+def translate(sentence)
+  words = sentence.split(" ")
+  pig_latin_words = words.map { |word| translate_word(word) }
+  pig_latin_words.join(" ")
+end
+
+def translate_word(word)
+  capitalized = (word[0] == word[0].upcase)
+  word = word.downcase
+  if word =~ /\A[aeiou]/
+    result = word + "ay"
+  elsif word =~ /\A(qu)/
+    result = word[2..-1] + "quay"
+  elsif word =~ /\A([^aeiou]*qu)/
+    match = word.match(/\A([^aeiou]*qu)(.*)/)
+    result = match[2] + match[1] + "ay"
+  else
+    match = word.match(/\A([^aeiou]+)(.*)/)
+    result = match[2] + match[1] + "ay"
+  end
```

Comment:
This works, but in Ruby you can avoid repeating `result =` by assigning the result of the `if` chain to the variable like so:

```suggestion
  result = if word =~ /\A[aeiou]/
    word + "ay"
  elsif word =~ /\A(qu)/
    word[2..-1] + "quay"
  elsif word =~ /\A([^aeiou]*qu)/
    match = word.match(/\A([^aeiou]*qu)(.*)/)
    match[2] + match[1] + "ay"
  else
    match = word.match(/\A([^aeiou]+)(.*)/)
    match[2] + match[1] + "ay"
  end
```